lib/ui/survey_question/survey_answer_views/emoticon_view.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/material.dart';
+import 'package:survey_flutter_ic/model/enum/emoticon_type.dart';
+import 'package:survey_flutter_ic/ui/survey_question/survey_answer_views/survey_answer_component_id.dart';
+import 'package:survey_flutter_ic/utils/dimension.dart';
+
+class EmoticonView extends StatefulWidget {
+  final EmoticonType type;
+  final Function(int) onSelect;
+
+  const EmoticonView({
+    super.key,
+    required this.type,
+    required this.onSelect,
+  });
+
+  @override
+  State<EmoticonView> createState() => _EmoticonViewState();
+}
+
+class _EmoticonViewState extends State<EmoticonView> {
+  int? selectedIndex;
+
+  List<Widget> get _likertButtons {
+    List<Widget> widgets = [];
+    widget.type.icons.asMap().forEach((index, icon) {
+      widgets.add(
+        GestureDetector(
```

Comment:
@ducbm051291 @thiennguyen0196 

Optional: There are multiple way to adopt tapping on a widget. By using `GestureDetector` widget or `InkWell` widget. You guys can try it out. Over here, I recommend you try the `InkWell` widget to see the differences between them:

For the `InkWell` widget, it has a nice ink-speading effect when tapping on it:

<img width="949" alt="image" src="https://github.com/nimblehq/flutter-ic-david-bruce/assets/23162627/1db1586e-45e6-474a-a6aa-f4fee86fde59">

Please check the full document [here](https://api.flutter.dev/flutter/material/InkWell-class.html) if you want.

So now, let's give it a try. 😄

Reply:
Amazing, let me give it a try